semantic-release/action.yaml

```diff
@@ -62,7 +68,27 @@ runs:
         node-version: "18"
     - name: Semantic-release
       id: semantic-release
-      run: node ${{ github.action_path }}/dist/index.js
+      run: |
+        if [[ ! -z "${{ inputs.override-github-ref-name }}" ]]; then
+
+          # if the override-github-ref-name is main or master, error
+          if [[ "${{ inputs.override-github-ref-name }}" == "main" || "${{ inputs.override-github-ref-name }}" == "master" ]]; then
+            echo "::error::Cannot override to main or master branch"
+            exit 1
+          fi
+          echo "::notice::Overriding github ref to ${{ github.ref }}"
+          export GITHUB_REF=refs/heads/${{ inputs.override-github-ref-name }}
+          export GITHUB_REF_NAME=${{ inputs.override-github-ref-name }}
+          export GITHUB_SHA=${{ inputs.override-github-sha }}
```

Comment:
For the future. Should we check that the sha belongs to the ref-name?

Reply:
hmm, should we just pull the sha from the current git branch?  Maybe we don't need to pass it in at all.